dm/worker/server_test.go

```diff
@@ -202,3 +202,81 @@ func (t *testServer) TestQueryError(c *C) {
 	c.Assert(resp.SubTaskError, HasLen, 1)
 	c.Assert(resp.SubTaskError[0].String(), Matches, `[\s\S]*mockSubtaskFail[\s\S]*`)
 }
+
+func (t *testServer) TestUnifyMasterBinlogPos(c *C) {
+	var (
+		pos1 = "(bin.000001, 3134)"
+		pos2 = "(bin.000001, 3234)"
+		pos3 = "(bin.000001, 3334)"
+		pos4 = "(bin.000001, 3434)"
+	)
+
+	// 1. should modify nothing
+	resp := &pb.QueryStatusResponse{
+		SubTaskStatus: []*pb.SubTaskStatus{{
+			Name:   "test",
+			Status: &pb.SubTaskStatus_Msg{Msg: "sub task not started"},
+		}},
+		RelayStatus: &pb.RelayStatus{
+			Stage: pb.Stage_Stopped,
+		},
+	}
+	resp2 := &pb.QueryStatusResponse{
+		SubTaskStatus: []*pb.SubTaskStatus{{
+			Name:   "test",
+			Status: &pb.SubTaskStatus_Msg{Msg: "sub task not started"},
+		}},
+		RelayStatus: &pb.RelayStatus{
+			MasterBinlog: pos1, RelayBinlog: pos1, RelayCatchUpMaster: true,
+		},
+	}
+	resp3 := &pb.QueryStatusResponse{
+		SubTaskStatus: []*pb.SubTaskStatus{{
+			Unit: pb.UnitType_Load,
+		}, {
+			Unit:   pb.UnitType_Sync,
+			Status: &pb.SubTaskStatus_Sync{Sync: &pb.SyncStatus{MasterBinlog: pos2, SyncerBinlog: pos2, Synced: true}},
+		}},
+		RelayStatus: &pb.RelayStatus{
+			Stage: pb.Stage_Stopped,
+		},
+	}
+
+	for _, r := range []*pb.QueryStatusResponse{resp, resp2, resp3} {
+		// clone resp
+		bytes, _ := r.Marshal()
+		originReps := &pb.QueryStatusResponse{}
+		err := originReps.Unmarshal(bytes)
+		c.Assert(err, IsNil)
+
+		unifyMasterBinlogPos(resp)
```

Comment:
`unifyMasterBinlogPos(r)`?

Reply:
addressed in https://github.com/pingcap/dm/pull/817/commits/2b09abe4daa0e85ccdf39a569a1e8bae80e0ad75